packages/sdk-codegen/src/typescript.gen.spec.ts

```diff
@@ -351,16 +384,75 @@ body: ICreateDashboardRenderTask`)
 }`)
     })
 
+    describe('special symbol names', () => {
+      interface HiFen {
+        'a-one': string
+        'a two': boolean
+        'a-three': number
+      }
+
+      it('handles special names in json', () => {
+        const json = `{"a-one":"one", "a two":true, "a-three":3}`
+        const actual: HiFen = JSON.parse(json)
+        expect(actual['a-one']).toEqual('one')
+        expect(actual['a two']).toEqual(true)
+        expect(actual['a-three']).toEqual(3)
+      })
+
+      it('reserves special names in method parameters', () => {
+        const method = apiTestModel.methods.me
+        const save = method.params[0].name
+        method.params[0].name = 'hi-test'
+        const actual = gen.declareMethod(indent, method)
+        method.params[0].name = save
+        const expected = `/**
+ * ### Get information about the current user; i.e. the user account currently calling the API.
+ *
+ * GET /user -> IUser
+ */
+async me(
+  /**
+   * @param {string} hi-test Requested fields.
+   */
+  'hi-test'?: string,
```

Comment:
interesting. how do you access this parameter inside the function?

Reply:
```typescript
method['hi-test']
```

Reply:
go figure